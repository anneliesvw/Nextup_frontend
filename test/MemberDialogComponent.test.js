import Vuex from 'vuex';
import { shallow, createLocalVue } from 'vue-test-utils';
import MemberDialogComp from '../src/components/groups/MemberDialog.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('MemberDialogComp', () => {
  let store;
  let actions;
  beforeEach(() => {
    actions = {
      addUserToGroup: jest.fn(),
      deleteUserFromGroup: jest.fn(),
    };
    store = new Vuex.Store({
      actions,
    });
  });

  it('dispatches addUserToGroup and deleteUserFromGroup when buttons clicked', () => {
    const wrapper = shallow(MemberDialogComp, {
      store,
      localVue,
    });
    wrapper.setProps({ activeGroup: { groupId: 1, users: [{ userId: 1 }] } });
    wrapper.find('.appended-input').trigger('click');
    wrapper.find('.appended-input').trigger('click');
    expect(actions.addUserToGroup.mock.calls).toHaveLength(2);
    wrapper.find('.member-remove').trigger('click');
    expect(actions.deleteUserFromGroup.mock.calls).toHaveLength(1);
  });
});
