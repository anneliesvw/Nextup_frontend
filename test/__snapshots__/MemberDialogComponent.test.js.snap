// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemberDialogComp should match snapshot 1`] = `
<el-dialog
  class="invite-member-dialog"
  title="Invite Others"
>
  <div
    class="invite-member-body"
  >
    <div
      class="invite-member-input"
    >
      <el-input
        placeholder="invite others by entering their email"
      >
        <el-button
          class="appended-input"
          slot="append"
          type="primary"
        >
          Invite
        </el-button>
      </el-input>
    </div>
     
    <div
      class="no-members-message"
    >
      
      
      
      <!---->
    </div>
  </div>
</el-dialog>
`;
